pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -893,54 +893,69 @@ func (c *controller) getVMStatus(getMachineStatusRequest *driver.GetMachineStatu
-		nodeName                = machine.Labels["node"]
-		nodeNotReadyDuration    = 5 * time.Minute
+		machine                                      = deleteMachineRequest.Machine
+		maxEvictRetries                              = int32(math.Min(float64(*c.getEffectiveMaxEvictRetries(machine)), c.getEffectiveDrainTimeout(machine).Seconds()/drain.PodEvictionRetryInterval.Seconds()))
+		pvDetachTimeOut                              = c.safetyOptions.PvDetachTimeout.Duration
+		timeOutDuration                              = c.getEffectiveDrainTimeout(deleteMachineRequest.Machine).Duration
+		forceDeleteLabelPresent                      = machine.Labels["force-deletion"] == "True"
+		nodeName                                     = machine.Labels["node"]
+		nodeNotReadyDuration                         = 5 * time.Minute
+		ReadonlyFilesystem      v1.NodeConditionType = "ReadonlyFilesystem"
 	)
 
 	if !isValidNodeName(nodeName) {
-		klog.Warningf("Skipping drain as nodeName is not a valid one for machine %q", machine.Name)
-		err = fmt.Errorf("Skipping drain as nodeName is not a valid one for machine. %s", machineutils.InitiateVMDeletion)
-		description = fmt.Sprintf("Skipping drain as nodeName is not a valid one for machine. %s", machineutils.InitiateVMDeletion)
+		message := "Skipping drain as nodeName is not a valid one for machine."
+		printLogInitError(message, &err, &description, machine)
 		skipDrain = true
 	} else {
+
 		for _, condition := range machine.Status.Conditions {
+
 			if condition.Type == v1.NodeReady {
-				if condition.Status != corev1.ConditionTrue && (time.Since(condition.LastTransitionTime.Time) > nodeNotReadyDuration) {
-					klog.Warningf("Skipping drain for NotReady machine %q", machine.Name)
-					err = fmt.Errorf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
-					description = fmt.Sprintf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
-					skipDrain = true
-				}
-				// break once the condition is found
-				break
+				nodeReadyCondition = condition
+			} else if condition.Type == ReadonlyFilesystem {
+				readOnlyFileSystemCondition = condition
 			}
 		}
+
+		if !isConditionEmpty(nodeReadyCondition) && (nodeReadyCondition.Status != corev1.ConditionTrue) && (time.Since(nodeReadyCondition.LastTransitionTime.Time) > nodeNotReadyDuration) {
+			message := "Skipping drain as machine is NotReady for over 5minutes."
+			printLogInitError(message, &err, &description, machine)
+			skipDrain = true
+		} else if !isConditionEmpty(nodeReadyCondition) && !isConditionEmpty(readOnlyFileSystemCondition) && nodeReadyCondition.Status == corev1.ConditionTrue && (readOnlyFileSystemCondition.Status == corev1.ConditionTrue) && (time.Since(readOnlyFileSystemCondition.LastTransitionTime.Time) > nodeNotReadyDuration) {
+			message := "Skipping drain as machine is Ready with ReadonlyFilesystem true for over 5minutes."
```

Comment:
If above assumption holds good. We can change this message to something like this?
```suggestion
			message := "Skipping drain as machine is in ReadonlyFilesystem for over 5minutes."
```